File: codeforces/462/c/c.go
///
// File:  c.go
// Author: ymiyamoto
//
// Created on Tue Mar 13 22:21:00 2018
//
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
	"strconv"
	"strings"
)

var N int
var arr []int

var reader = bufio.NewReaderSize(os.Stdin, 1000000)

func readLine() string {
	buf := make([]byte, 0, 1000000)
	for {
		l, p, _ := reader.ReadLine()
		buf = append(buf, l...)
		if !p {
			break
		}
	}
	return string(buf)
}

func main() {
	fmt.Scan(&N)
	fmt.Println()

	arr = make([]int, N)
	for i, v := range strings.Split(readLine(), " ") {
		arr[i], _ = strconv.Atoi(v)
	}

	sort.Ints(arr)

	var count int64
	for i, v := range arr {
		count += int64((2 + i) * v)
	}
	count -= int64(arr[len(arr)-1])

	fmt.Println(count)
}

File: codeforces/462/c/c.cpp
///
// File:  c.cpp
// Author: ymiyamoto
//
// Created on Wed Mar 14 22:07:06 2018
//
#include <bits/stdc++.h>
using namespace std;

#define rep(_n, _init, _N) for (int32_t _n = _init; _n < (int32_t)(_N); _n++)
#define rrep(_n, _N, _end) for (int32_t _n = (int32_t)(_N); _n >= _end; _n--)
#define iceil(_x, _y) (((_x) + (_y)-1) / (_y))
#define ifloor(_x, _y) ((_x) / (_y))

static const double pi = 2 * asin(1);
static const double precision = 10e-9;

static int32_t N;

int32_t main()
{
  cin >> N;

  vector<uint64_t> arr(N, 0);
  rep(i, 0, N)
  {
    cin >> arr[i];
  }

  sort(arr.begin(), arr.end());

  uint64_t count = 0;
  rep(i, 0, N)
  {
    count += (i + 2) * arr[i];
  }
  count -= arr[arr.size() - 1];

  cout << count << endl;

  return 0;
}
